### PR TITLE
Added Codeowners to Clamshell.

### PR DESCRIPTION
**Pull request by nelsonjhk**
Github allows for auto-assigning code owners to relevant codebases. This file assigns maintainers to pull requests for final review.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Codeowners file for auto-assignment of experienced developers for code review
+
+*	@kashomon @mhoak
+
+# Auto-assignment of relevant developers for code review
+
+core/sgf/	@kashomon @mhoak @TrevorPeters
```
Comment (artasparks) [inline .github/CODEOWNERS:7]: Are directory assignment rules inherited? 

I.e., does this also get the * rules by default? In which case we could just put TrevorPeters?

**Review by artasparks — APPROVED** [review]

**Comment by artasparks** [comment]
Fixes #140

File: .github/CODEOWNERS (hunk 1 above)
Comment (nelsonjhk) [reply .github/CODEOWNERS:7]: My understanding is that the lowest level takes precedence, in which case only TrevorPeters would be assigned.

**Comment by artasparks** [comment]
LGTM.

**Pull request merged**